### PR TITLE
rewrite use_bbi with built-in R functionality

### PR DESCRIPTION
**Pull request by kyleam**
Prompted by gh-446, I started  looking over `use_bbi`.  This series rewrites `use_bbi` to hopefully make it more portable, as well resolve others issues noted in the second commit message.  It also incidentally fixes gh-409.

### CODE REVIEW

File: R/use-bbi.R [hunk 1]
```diff
@@ -225,89 +219,52 @@ build_bbi_install_path <- function() {
...
+    message(glue("Copying bbi to '{.path}'"))
   }
+  file.copy(exe, .path, overwrite = TRUE)
 
-  # extract dir name that tar will unzip to
-  tar_basename <- stringr::str_replace(basename(.bbi_url), "\\..+$", "")
-
-  # construct commands
-  # WE USE ONLY THE install_dir, because I can't get a straight answer about Windows temp dirs
-  temp_dir <- file.path(dirname(.path), "TEMP")
```
Comment (kyleam) [inline R/use-bbi.R:298]: With a quick search, I didn't find anything about `tempfile` not working on Windows.  @dpastoor, if you have a moment, can you check whether `tempfile` seems to behave sensibly on Windows for you (not in the context of this PR, just a plain call to `tempfile`)?

**Comment by seth127** [comment]
I looked over this and love it, but I haven't gotten to run it or formally review it. @dpastoor if you want to review this go for it (I think you wrote most of this original code that's being refactored). If not, I can get to it middle of next week.

**Review by seth127 — APPROVED** [review]
Looks good and works for me on both Mac and Linux. Sidenote: I'm gonna make another PR off this that cleans up the (currently misleading) message that this prints.

File: R/utils.R [hunk 2]
```diff
@@ -491,3 +491,16 @@ suppressSpecificWarning <- function(.expr, .regexpr) {
   })
 }
 
+#' Call `utils::download.file()`, retrying once on failure.
+#'
+#' @param ... arguments to pass to `utils::download.file()`
+#' @keywords internal
+download_with_retry <- function(...) {
```
[the remark this answers is not in the document]
Comment (kyleam) [reply R/utils.R:498]: In general, I don't think it's unreasonable to guard against transient network failure, but really it's just retaining the previous behavior.  From the commit message:

```
utils: add download.file() helper that retries once

The next commit will replace the system wget calls in use_bbi() with
calls to download.file().  Those wget calls use --tries=2, which leads
to wget trying the download twice, sleeping one second after a
failure.

Add a download.file() wrapper that hard codes that behavior [*].
Although it seems natural to let the caller specify things like the
number of retries, punt on that until there's a need.

[*] ... glossing over the fact that wget does _not_ retry for 'fatal
    errors like "connection refused" or "not found" (404)'
```

That `--tries=2` (along with a timeout) was added in b2f6f1d (Fixing use_bbi hang, 2020-06-22).

**Comment by kyleam** [comment]
> Looks good and works for me on both Mac and Linux. 
Thanks for testing it out.

> Sidenote: I'm gonna make another PR off this that cleans up the (currently misleading) message that this prints.

Such mystery and suspense :). If it's closely related to this PR, feel free to push it here if you'd prefer.

**Pull request merged**